src/metatensor/models/alchemical_model/model.py

```diff
@@ -0,0 +1,296 @@
+        ps_input_size = self.soap_features_layer.num_features
+
+        self.layer_norm = LayerNorm(ps_input_size)
+        vex_calculator = (
+            self.soap_features_layer.spex_calculator.vector_expansion_calculator
+        )
+        contraction_layer = vex_calculator.radial_basis_calculator.combination_matrix
+        self.embedding = AlchemicalEmbedding(
+            unique_numbers=self.all_species,
+            num_pseudo_species=self.num_pseudo_species,
+            contraction_layer=contraction_layer,
+        )
+
+        num_hidden_layers = hypers["bpnn"]["num_hidden_layers"]
+        num_neurons_per_layer = hypers["bpnn"]["num_neurons_per_layer"]
+        activation_function = hypers["bpnn"]["activation_function"]
+        if activation_function == "SiLU":
+            self.activation_function = SiLU()
+        else:
+            raise ValueError(
+                f"Activation function {activation_function} not supported."
+            )
+
+        layer_size = [ps_input_size] + [num_neurons_per_layer] * num_hidden_layers
+        bpnn_layers = []
+        for layer_index in range(1, len(layer_size)):
+            bpnn_layers.append(
+                MultiChannelLinear(
+                    in_features=layer_size[layer_index - 1],
+                    out_features=layer_size[layer_index],
+                    num_channels=self.num_pseudo_species,
+                    bias=False,
+                )
+            )
+            bpnn_layers.append(self.activation_function)
+
+        n_inputs_last_layer = layer_size[-1]
+        n_outputs_last_layer = 1
+
+        self.bpnn = torch.nn.ModuleList(bpnn_layers)
+
+        self.last_layers = torch.nn.ModuleDict(
+            {
+                output_name: MultiChannelLinear(
+                    in_features=n_inputs_last_layer,
+                    out_features=n_outputs_last_layer,
+                    num_channels=self.num_pseudo_species,
+                    bias=False,
+                )
+                for output_name in capabilities.outputs.keys()
+            }
+        )
+
+    def requested_neighbors_lists(
+        self,
+    ) -> List[NeighborsListOptions]:
+        return [
+            NeighborsListOptions(
+                model_cutoff=self.cutoff_radius,
+                full_list=True,
+            )
+        ]
+
+    def forward(
+        self,
+        systems: List[System],
+        outputs: Dict[str, ModelOutput],
+        selected_atoms: Optional[Labels] = None,
+    ) -> Dict[str, TensorMap]:
+        if selected_atoms is not None:
+            raise NotImplementedError(
+                "Alchemical Model does not support selected atoms."
+            )
+
+        for requested_output in outputs.keys():
+            if requested_output not in self.capabilities.outputs.keys():
+                raise ValueError(
+                    f"Requested output {requested_output} is not within "
+                    "the model's capabilities."
+                )
```

Comment:
this is already checked by `MetatensorAtomisticModel`. If we need to duplicate the check could you please add a comment about why?